core/src/main/java/com/graphhopper/reader/osm/OSMXMLHelper.java

```diff
@@ -105,7 +104,13 @@ private static void readMembers(ReaderRelation rel, XMLStreamReader parser) thro
 
     public static Member createMember(XMLStreamReader parser) {
         String typeName = parser.getAttributeValue(null, "type");
-        int type = TYPE_DECODE.indexOf(typeName.charAt(0));
+        ReaderElement.Type type = ReaderElement.Type.NODE;
+        if (typeName.startsWith("w")) {
+        	type = ReaderElement.Type.WAY;
+        }
+        else if (typeName.startsWith("r")) {
+        	type = ReaderElement.Type.RELATION;
+        }
```

Comment:
This change might affect performance, but I doubt it. If it did we will notice in our performance metrics and can still adjust after merging this. Please also note that the code formatting seems off (we use the default formatting of the IntelliJ IDE). But I'll just fix this in master.

Reply:
Anything even remotely performance critical will not use XML anyway :wink:

Reply:
Ah, I see this is only used for xml.